src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -1943,6 +1943,110 @@ internal void IncrementPasteOffset()
             this.currentPasteOffset = (this.currentPasteOffset + PasteOffsetStep) % PasteOffsetMax;
         }
 
+        #region [Nodes Info]
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings
+        /// </summary>
+        public bool HasWarnings
+        {
+            get { return Nodes.Any(n => n.State == ElementState.Warning || n.State == ElementState.PersistentWarning); }
+        }
+
+        /// <summary>
+        /// Boolean indicates if the workspace run with warnings with no Geometry
+        /// </summary>
+        public bool HasWarningsWithNoGeometry
+        {
+            get { return Nodes.Any(n => (n.State == ElementState.Warning || n.State == ElementState.PersistentWarning) && !n.Category.StartsWith("Geometry.")); }
```

Comment:
This just means you're filtering all geometry nodes with ANY warning, that doesn't seem right.

Reply:
All the nodes with Warning status that not belong to the Geometry .Category actually.

Reply:
`HasNoneGeometryRelatedWarnings` instead for clarity?

Reply:
@jesusalvino ^^ please